test/conversion_test.go

```diff
@@ -0,0 +1,794 @@
+          exit 255
+  timeout: 60s
+  podTemplate:
+    securityContext:
+      fsGroup: 65532
+  workspaces:
+  - name: password-vault
+  finally:
+  - name: echo-status
+    params:
+      - name: echoStatus
+        value: "status"
+    taskSpec:
+      params:
+        - name: echoStatus
+          type: string
+      steps:
+        - name: verify-status
+          image: ubuntu
+          script: |
+            if [ $(params.echoStatus) == "Succeeded" ]
+            then
+              echo " Good night! echoed successfully"
+            fi
+`
+
+	v1beta1PipelineRunYaml = `
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  pipelineRef:
+    name: pr
+    bundle: test-bundle
+  params:
+  - name: STRING_LENGTH
+    value: 1
+    type: string
+  serviceAccountName: default
+  workspaces:
+  - name: password-vault
+    secret:
+      secretName: secret-password
+  timeout: 60s
+  resources:
+  - name: source-repo
+    resourceRef:
+      name: skaffold-git-output-pipelinerun
+`
+
+	v1beta1PipelineRunExpectedYaml = `
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  pipelineRef:
+    resolver: "bundles"
+    kind: Task
+    params:
+    - name: bundle
+      type: string
+      value: "test-bundle"
+    - name: name
+      type: string
+      value: pr
+    - name: kind
+      type: string
+      value: Task
+  params:
+  - name: STRING_LENGTH
+    value: 1
+    type: string
+  timeouts:
+    pipeline: 60s
+  workspaces:
+  - name: password-vault
+    secret:
+      secretName: secret-password
+  serviceAccountName: "default"
+  resources:
+  - name: source-repo
+    resourceRef:
+      name: skaffold-git-output-pipelinerun
+`
+
+	v1PipelineRunYaml = `
+metadata:
+  name: %s
+  namespace: %s
+  annotations: {
+    tekton.dev/v1beta1Resources: '[{"name":"source-repo","resourceRef":{"name":"skaffold-git-output-pipelinerun"}}]'
+  }
+spec:
+  pipelineRef:
+    resolver: "bundles"
+    kind: Task
+    params:
+    - name: bundle
+      type: string
+      value: "test-bundle"
+    - name: name
+      type: string
+      value: pr
+    - name: kind
+      type: string
+      value: Task
+  params:
+  - name: STRING_LENGTH
+    value: 1
+    type: string
+  timeouts:
+    pipeline: 60s
+  workspaces:
+  - name: password-vault
+    secret:
+      secretName: secret-password
+  taskRunTemplate: {
+    serviceAccountName: "default"
+  }
+`
+)
+
+// TestTaskCRDConversion first creates a v1beta1 Task CRD using v1beta1Clients and
+// requests it by v1Clients to compare with v1 if the conversion has been
+// correctly executed by the webhook. And then it creates the v1 Task CRD using v1Clients
+// and requests it by v1beta1Clients to compare with v1beta1.
+func TestTaskCRDConversion(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	t.Parallel()
+
+	c, namespace := setup(ctx, t)
+	knativetest.CleanupOnInterrupt(func() { tearDown(ctx, t, c, namespace) }, t.Logf)
+	defer tearDown(ctx, t, c, namespace)
```

Comment:
Curious why both `cleanuponinterrupt` and `defer tearDown` are needed?

Reply:
My take is `cleanuponinterrupt` only cleans up the namespace when the test fails https://github.com/tektoncd/pipeline/blob/main/vendor/knative.dev/pkg/test/cleanup.go#L57-L65 while defer tearDown works at the success of the test case? 

Also what's used in the test folders right now as in https://github.com/tektoncd/pipeline/blob/main/test/artifact_bucket_test.go#L61-L62 are deprecated. Created [issue](https://github.com/tektoncd/pipeline/issues/5728) to track down